enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,910 @@
+
+This eventually will allow the Machine API resources to be deleted and have no effect on the running system.
+
+Instead, we could always synchronise deletion events.
+This would mean that any delete of a resource is always mirrored no matter the circumstances.
+
+This would mean that, to allow users to clean up Machine API resources, we would need to introduce some “escape hatch”,
+likely an annotation, that a user could set on the resource to allow it to be deleted without also deleting the Cluster API equivalent.
+
+This could lead to users believing that, because the Cluster API resource is authoritative, deleting the Machine API resource is acceptable.
+Instead, it would result in deleting a Machine which they had not intended to do.
+This could be catastrophic to clusters if users accidentally delete a control plane Machine,
+therefore we need to make this as simple as possible, by having the single authoritative switch indicate the behaviour of deletion,
+there is less room for error on the customers part.
+
+### Use webhooks to synchronise annotations
+
+Instead of having each controller look up the annotation on the Machine API resource,
+we could use webhooks on both Machine API and Cluster API resources to ensure that only one of the resources is authoritative at any time.
+
+With this approach, the authoritative annotation existing on a resource would indicate that it is the authoritative version, and therefore should be reconciled. 
+Webhooks would be in place to prevent the annotation from being added to both Machine API and Cluster API mirrors at the same time.
+
+To move between the two versions, the user would first have to remove the annotation from one (in which case no reconciliation happens anywhere), and then add it to the other to make it authoritative.
+
+This approach may seem to be less complex, but has the potential to cause issues if, for some reason, the webhooks do not work correctly (eg a change is made while the webhook is being restarted).
+
+To avoid this potential, our approach suggests to have the authoritative annotation exist in only one location, on the Machine API resource.
+
+### Cluster wide migration of resources
+
+In this alternative implementation, the switch to allow migration between Cluster API and Machine API would be at the cluster level, rather than at the individual resource level.
+This alternative prevents users from testing migrations on individual resources and also prevents users from leveraging new Cluster API features until all of their Machine API machines are migratable.
+Since we understand that some feature parity may not be present in the initial releases of this feature, we do not want to block users from using the new features, while waiting for the entire cluster to be ready to migrate.
+
+## Infrastructure Needed
+
+A new repository `github.com/openshift/machine-api-synchronisation-operator` will be needed to complete this project.
```

Comment:
Just noticed this line. Does that mean the "operator" referred to under `## Graduation Criteria` above is a new Operator specifically to handle this?

Reply:
@jeana-redhat Yeah, there will be an operator to do the synchronization/copy of the data, and one to manage which API is authoritative.

Reply:
maybe it can use `machine-api-sync-operator` so no one has to write it out or choose an English variant :wink: